test/test_network.h

```diff
@@ -485,7 +487,7 @@ TEST(network, gradient_check6) {  // sigmoid - cross-entropy
 
 TEST(network, gradient_check7) {  // leaky-relu - mse
   typedef mse loss_func;
-  typedef leaky_relu activation;
+  typedef leaky_relu_layer activation;
```

Comment:
You might want to use `using` feature here
proposal:use using instead of typedef